### PR TITLE
MTN set get_next as a regular method

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**

### CODE REVIEW

**Comment by mathurinm** [comment]
failure seems due to R solver, do you confirm @Badr-MOUFAD ?

File: solvers/cd.py [hunk 1]
```diff
@@ -66,7 +66,6 @@ def run(self, n_iter):
             L = (self.X ** 2).sum(axis=0)
             self.w = self.cd(self.X, self.y, self.lmbd, L, n_iter)
 
-    @staticmethod
```
Comment (mathurinm) [inline solvers/cd.py:69]: why this change?

File: solvers/cd.py [hunk 2]
```diff
@@ -83,7 +82,6 @@ def cd(X, y, lmbd, L, n_iter):
                     R += diff * X[:, j]
         return w
 
-    @staticmethod
```
Comment (mathurinm) [inline solvers/cd.py:86]: same

File: solvers/julia_pgd.jl [hunk 3]
```diff
@@ -1,21 +1,22 @@
 using Core
 using LinearAlgebra
 
-st(w::Vector{Float64}, t::Float64) = @. sign(w) * max(abs(w) - t, 0.)
+st(w:: Vector{Float64}, t: : Float64) = @. sign(w) * max(abs(w) - t, 0.)
```
Comment (mathurinm) [inline solvers/julia_pgd.jl:4]: are these necessary or an artifact ?

File: solvers/r_pgd.R [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
+    p < - ncol(X)
+    parameters < - numeric(p)
+    step_size < - 1 / norm(X, "2") ** 2
+    for (i in 1: n_iter) {
         # Compute the gradient
-        grad <- crossprod(-X, (Y - X %*% parameters))
+        grad < - crossprod(-X, (Y - X % * % parameters))
         # # Update the parameters
-        parameters <- St(step_size * lambda, parameters - step_size * grad)
+        parameters < - St(step_size * lambda, parameters - step_size * grad)
```
Comment (mathurinm) [inline solvers/r_pgd.R:34]: not sure this should be touched

**Comment by Badr-MOUFAD** [comment]
we should expect the tests to fail on the release branch, right @mathurinm?

**Review by mathurinm — APPROVED** [review]
